rfcs/RFC-Upload-From-Dropbox.md

```diff
@@ -0,0 +1,358 @@
+| --------------------- | ----------------------------------------------------------------------------------------------------------------------------------------------------- |
+| Link                  | a shared link to the file                                                                                                                             |
+| submission_id         | identifies the submission                                                                                                                             |
+| _Optional_ dataset_id | identifies the dataset being uploaded. Used to check the status of the upload.                                                                        |
+| _Optional_ overwrite  | A boolean, indicating if the user would like to overwrite the file that was uploaded. The upload status must be complete for the dataset_id provided. |
+
+**Response:**
+
+| Key        | Description                                                                |
+| ---------- | -------------------------------------------------------------------------- |
+| dataset_id | identifies the data set the file will be associated with after validation. |
+| status     | Provides the current status of the upload                                  |
+
+**Error Responses:**
+
+| Code | Description                                                                                                      |
+| ---- | ---------------------------------------------------------------------------------------------------------------- |
+| 401  | if dataset_id or submission_id does not exist, or if the user does not own the submission or upload in-progress. |
+| 400  | if the file type is invalid                                                                                      |
+
+#### Delete submission/{submission_id}/upload
+
+Cancels an existing upload job. Any data that has started to upload is removed, and the upload status is changed to
+cancel pending until the job has been cleared up.
+
+**Request:**
+
+| Parameter     | Description                           |
+| ------------- | ------------------------------------- |
+| submission_id | identifies the submission             |
+| dataset_id    | identifies the dataset being canceled |
+
+**Response:**
+
+| Key    | Description                                |
+| ------ | ------------------------------------------ |
+| status | Provides the current status of the upload. |
+
+**Error Responses:**
+
+| Code | Description                                                                                                             |
+| ---- | ----------------------------------------------------------------------------------------------------------------------- |
+| 401  | if dataset_id or submission_id does not exist, or if the user does not own the submission or upload in-progress upload. |
```

Comment:
I'd reserve 401 for "user does not own the submission ..."
For dataset_id and submitssion_id does not exist, that could be 404 or 400.

Reply:
Using 404 allows a user who does not own a submission to probe our system for ids. They would get a 404 if the id doesn't exists and a 401 if it exists and they don't have access. It's better to just return 401 to prevent probing. 

I think 400 would imply that there was a mistake in their parameters.

Do you have a more compelling reason for using 404 or 400?

Reply:
Thanks for the explanation.   That makes sense.